src/pages/Admin/Admin.tsx

```diff
@@ -284,15 +285,48 @@ const EventsComponent: React.FC<VenueDetailsPartProps> = ({ venue }) => {
   const [showCreateEventModal, setShowCreateEventModal] = useState(false);
   const [showDeleteEventModal, setShowDeleteEventModal] = useState(false);
   const [editedEvent, setEditedEvent] = useState<WithId<VenueEvent>>();
-  //filter out events that have already finished
-  const filteredEvents = events?.filter(
-    (ev) =>
-      (ev.start_utc_seconds + ev.duration_minutes * 60) * 1000 > Date.now()
-  );
+  const [filterPastEvents, setFilterPastEvents] = useState(false);
+  const [filterText, setFilterText] = useState("");
+
+  let filteredEvents = events;
+  filterPastEvents
+    ? (filteredEvents = events?.filter(
+        (ev) =>
+          (ev.start_utc_seconds + ev.duration_minutes * 60) * 1000 > Date.now()
+      ))
+    : (filteredEvents = events);
```

Comment:
this can be written more tersely. const filteredEvents = filteredPastEvents ? ...

Also wrap in useMemo